lib/src/cli/install/install_command.dart

```diff
@@ -237,195 +125,82 @@ class InstallCommand extends Command<void> {
     if (packageConfig == null) {
       throw Exception("Package configuration not found. "
-          "Run the 'dart run $packageName install` command from the root directory of your application");
+          "Run the 'dart run $_packageName install` command from the root directory of your application");
     }
 
-    final realmDartPackage = packageConfig.packages.where((p) => p.name == packageName).firstOrNull;
+    final realmDartPackage = packageConfig.packages.where((p) => p.name == _packageName).firstOrNull;
     if (realmDartPackage == null) {
-      throw Exception("$packageName package not found in dependencies. Add $packageName package to your pubspec.yaml");
+      throw Exception("$_packageName package not found in dependencies. Add $_packageName package to your pubspec.yaml");
     }
 
     if (realmDartPackage.root.scheme != 'file') {
-      throw Exception("$packageName package uri ${realmDartPackage.root} is not supported. Uri should start with file://");
+      throw Exception("$_packageName package uri ${realmDartPackage.root} is not supported. Uri should start with file://");
     }
 
     final realmPackagePath = path.join(realmDartPackage.root.toFilePath(), "pubspec.yaml");
     return realmPackagePath;
   }
 
-  Future<Pubspec> parseRealmPubspec(String path) async {
+  Future<Pubspec> _parseRealmPubspec(String path) async {
     try {
       var realmPubspecFile = await File(path).readAsString();
       final pubspec = Pubspec.parse(realmPubspecFile);
-      if (pubspec.name != packageName) {
-        throw Exception("Unexpected package name `${pubspec.name}` at $path. Realm install command expected package `$packageName`");
+      if (pubspec.name != _packageName) {
+        throw Exception("Unexpected package name `${pubspec.name}` at $path. Realm install command expected package `$_packageName`");
       }
 
       return pubspec;
     } on Exception catch (e) {
-      throw Exception("Error parsing package `$packageName` pubspect at $path. Error $e");
+      throw Exception("Error parsing package `$_packageName` pubspec at $path. Error $e");
     }
   }
 
   @override
   FutureOr<void>? run() async {
-    await runInternal();
-    print("Realm install comamnd finished.");
-  }
+    _options = parseOptionsResult(argResults!);
+    _validateOptions();
 
-  FutureOr<void>? runInternal() async {
-    options = parseOptionsResult(argResults!);
-    validateOptions();
+    final realmPackagePath = await _getRealmPackagePath();
+    final realmPubspec = await _parseRealmPubspec(realmPackagePath);
 
-    if (await skipInstall()) {
-      print("Realm install command started from within the realm-dart repo or one of the realm packages is a path dependency or has a path dependency override."
-          " Skipping install.");
+    if (realmPubspec.publishTo == 'none') {
+      print("Referencing $_packageName@${realmPubspec.version} which hasn't been published. Skipping download.");
```

Comment:
This error message is misleading. When we publish a version we are going to keep the `publish_to: none` in the repo.

Reply:
This is intentional - we don't want to download binaries if we're working with the dart repo. When we upload the packages to S3, we remove the `publish_to: none`, so if you download a package from S3, you can reference it by path and run the install command to download the binaries.

Reply:
Not sure what do you mean. How are we uploading the packages to S3. `dart pub publish` does not do that.

Reply:
I added a discussion point for this. I think we can handle this use case better